src/smt/theory_str_noodler/formula_preprocess.cpp

```diff
@@ -1548,6 +1548,36 @@ namespace smt::noodler {
                 ret.push_back(bt);
             }
         }
+
+        // if (ret.empty()) return ret;
+
+        // int conseq_literals = 0;
+        // bool should_compress = false;
+        // for (const BasicTerm& term : ret) {
+        //     if (term.is_literal()) conseq_literals += 1;
+        //     if (conseq_literals >= 2) should_compress = true;
+        // }
+
+        // if (should_compress) {
+        //     Concat new_concat;
+        //     auto ret_iterator = ret.begin();
+        //     new_concat.push_back(*ret_iterator);
+        //     ret_iterator++;
+
+        //     for (; ret_iterator != ret.end(); ret_iterator++) {
+        //         int last_idx_in_new_concat = new_concat.size() - 1;
+        //         const BasicTerm& current_item = *ret_iterator;
+        //         if (new_concat.at(last_idx_in_new_concat).is_literal() && current_item.is_literal()) {
+        //             // @Optimize(mhecko): If we compute the buffer size beforehand, we might get away with doing less
+        //             //                    allocations -- should be faster.
+        //             zstring new_literal = new_concat[last_idx_in_new_concat].get_name() + current_item.get_name();
+        //             new_concat[last_idx_in_new_concat] = BasicTerm(BasicTermType::Literal, new_literal);
+        //         }
+        //     }
+
+        //     return new_concat;
+        // }
+
```

Comment:
Delete this if it is not gonna be used (or add comment why to keep).

Reply:
I have added a comment about what was the commented out code trying to solve (so that it can be addressed in the future). The commented-out code has been removed.